_source/2016-09-09-mlr-loves-OpenML.Rmd

```diff
@@ -0,0 +1,144 @@
+and machine learning experiments.
+
+At the time of writing this blog I am in Eindoven at an [OpenML
+workshop](http://openml2016dev.openml.org/), where developers and scientists
+meet to work on improving the project. Some of these people are R users and they (we)
+are developing an [R package](https://github.com/openml/openml-r) that
+communicates with the OpenML platform.
+
+
+![](../images/mlr_loves_openml.png) 
+
+## OpenML in R
+The OpenML R package can list and download data sets and machine
+learning tasks (prediction challenges).  In R one can run algorithms on the
+these data sets/tasks and
+then upload the results to OpenML. After successful uploading, the website shows how well the
+algorithm performs.  To run the algorithm on a given task the OpenML R package
+builds on the [mlr](https://github.com/mlr-org/mlr) package. mlr understands
+what a task is and can run learners on that task. So all the OpenML package
+needs to do is convert the OpenML objects to objects mlr understands and then
+mlr deals with the learning.
+
+
+## A small case study
+
+We want to create a little study on the [OpenML
+website](http://www.openml.org/), in which we compare different types of Support
+Vector Machines.  The study gets an ID assigned to it, which in our case is 27.
+We use the function ksvm (with different settings of the function argument type)
+from package kernlab, which is integrated in mlr ("classif.ksvm").
+
+![](../images/openml_screenshot_study.png) 
+
+
+For details on installing and setting up the OpenML R package please see the
+[guide](https://github.com/openml/openml-r) on GitHub.
+
+Let's start conducting the study:
+
+- Load the packages and list all tasks which have between 100 and 500
+  observations. 
+```{r, eval=TRUE, message=FALSE}
+library("OpenML")
+library("mlr")
+library("farff")
+```
+
+```{r}
+dsize = c(100, 500)
+taskinfo_all = listOMLTasks(number.of.instances = dsize)
+```
+
+- Select all supervised classification tasks that do 10-fold cross-validation
+  and choose only one task per data set. To keep the study simple and fast to compute, 
+  select only the first three tasks.
+```{r}
+taskinfo_10cv = subset(taskinfo_all, task.type == "Supervised Classification" & 
+                    estimation.procedure == "10-fold Crossvalidation" &
+                    evaluation.measures == "predictive_accuracy" &
+                    number.of.missing.values == 0 &
+                    number.of.classes %in% c(2, 4))
+taskinfo = taskinfo_10cv[1:3, ]
+
+```
+
+- Create the learners we want to compare.
+```{r}
+lrn.list = list(
+  makeLearner("classif.ksvm", type = "C-svc"),
+  makeLearner("classif.ksvm", type = "kbb-svc"),
+  makeLearner("classif.ksvm", type = "spoc-svc")
+)
+```
+
+- Run the learners on the three tasks.
+```{r}
+grid = expand.grid(task.id = taskinfo$task.id, 
+                   lrn.ind = seq_along(lrn.list))
+
+runs = lapply(seq_row(grid), function(i) {
```

Comment:
not sure what you guys are talking about here? can we make the sentences a bit longer and precise? maybe with active hints?